code/game/gamemodes/infection/infection.dm

```diff
@@ -0,0 +1,69 @@
+/datum/game_mode/infection
+	name = "infection"
+	config_tag = "infection"
+	report_type = "infection"
+	false_report_weight = 10
+	required_players = 40 // this is an all out station war
+	required_enemies = 2
+	recommended_enemies = 5
+	antag_flag = ROLE_INFECTION
+	enemy_minimum_age = 14 // these should be somewhat experienced players for an all out war mode
+
+	announce_span = "danger"
+	announce_text = "An infectious core is approaching on a meteor in an attempt to consume the station!\n\
+	<span class='danger'>Infection</span>: Destroy the defensive beacons to consume the station.\n\
+	<span class='notice'>Crew</span>: Defend long enough until you find a way to destroy the core."
+
+	var/list/pre_spores = list()
+
+	var/commander_datum_type = /datum/antagonist/infection
+
+	var/spore_datum_type = /datum/antagonist/infection/spore
+
+/datum/game_mode/infection/pre_setup()
+	if(!GLOB.infection_spawns.len)
```

Comment:
why is this a global list?

Reply:
It’s the spawn positions which are waypoints in the map. It’s used by the antagonist to spawn in the powers file. Similar to what’s done here https://github.com/tgstation/tgstation/blob/25f18ff5232f6db4ec6806535eed4a98e11e984b/code/modules/antagonists/nukeop/nukeop.dm#L97